builder/builder.go

```diff
@@ -106,7 +109,7 @@ type BuildOpt struct {
 	Logbus                     *logbus.Bus
 	MainTargetDetailsFunc      func(earthfile2llb.TargetDetails) error
 	Runner                     string
-	CloudStoredAuthProvider    cloudauth.ProjectBasedAuthProvider
+	ProjectAdder               ProjectAdder
```

Comment:
wondering about the field name - are we sure the previous name (or some variation of authProvider) is not clearer?

Reply:
Since the only method that was ever used was `AddProject`, I don't think that there was any point in declaring that it's an auth provider - because nothing really prevents us from using an entirely different type (as long as it has an `AddProject` method). I had to change the name because it's no longer `CloudStoredAuthProvider` - the type we're passing in is now  the `MultiAuthProvider`, so naming it something about `cloud` is misleading.

I don't really think `ProjectAdder` is very descriptive, but it's probably good enough, I think?

Reply:
I didn't even notice that was the only method used, sgtm!